pkgs/by-name/on/one-click-backup/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+, ninja
+, qt6
+, extra-cmake-modules
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "one-click-backup";
+  version = "1.2.2.1";
+
+  src = fetchFromGitLab {
+    owner = "dev-nis";
+    repo = "nis-one-click-backup-qt";
+    rev = finalAttrs.version;
+    hash = "sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=";
```

Comment:
```
error: hash mismatch in fixed-output derivation '/nix/store/c4v512z51l29bh34avhq53hbbjgh4ak3-source.drv':
         specified: sha256-JbgmfKEpKRj/ph8dkW0IRulmAhk6Dwv/1Z4j3hUgPXk=
            got:    sha256-F+gA+Z4gZoNJYdy28uIjqiJcwcNsyUzl6BXsiIZO0gE=
```

```suggestion
    hash = "sha256-F+gA+Z4gZoNJYdy28uIjqiJcwcNsyUzl6BXsiIZO0gE=";
```

Reply:
I knew I had to change that, but for some reason I don't understand, nix doesn't complain about this on my build machine. That's really strange.

Reply:
Does nix somehow cache the source for builds?

Reply:
It doesn't complain because that hash is still present on your system, from a previous/different build. Hash exists -> no need to build, just use what it points at.

Try building `one-click-backup.src` with `--check`, that should fail.

Reply:
Ah, that makes sense. Thank you for the explanation!